src/main/java/org/apache/commons/io/filefilter/AbstractFileFilter.java

```diff
@@ -37,7 +37,13 @@
      * @return true if this file matches the test
      */
     @Override
+    @SuppressWarnings("nullness:argument.type.incompatible")
     public boolean accept(final File file) {
+        // File.getParentFile() can return null, and null is not documented
+        // as a permitted file argument for FilenameFilter.accept(File,String).
+
+        // However following code does not lead to any Exception at run time
```

Comment:
@mernst  If method is not overridden, the program runs into Infinite loop. I have tested this behaviour, there was no null pointer exception, but stackoverflow error.   [AbstractFileFilter -reference](https://commons.apache.org/proper/commons-io/javadocs/api-2.2/org/apache/commons/io/filefilter/AbstractFileFilter.html)